src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -791,12 +791,24 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                     secureConnection = false;
                 }
 
+                // Try to use the requested version if a known/supported version was explicitly requested.
```

Comment:
It's seem too small at this point to add another method for just that. But I will re-consider later if this code block grows further.